python/mujinzmqclient/zmqclient.py

```diff
@@ -293,15 +293,18 @@ def port(self):
     def _CheckCallerThread(self, context=None):
         """Catch bad callers who use zmq client from multiple threads and cause random race conditions.
         """
-        callerthread = repr(threading.current_thread())
-        oldcallerthread = self._callerthread
-        oldcallercontext = self._callercontext
-        self._callerthread = callerthread
+        callerthread = threading.current_thread()
+        if self._callerthreadref is not None:
+            oldcallerthread = self._callerthreadref() # Get local ref for thread-safety.
+            oldcallercontext = self._callercontext
```

Comment:
you only need to do this once `oldcallerthread.native_id != callerthread.native_id` is met?

Reply:
I guess `oldcallercontext`  is not required at all. Removed